core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -505,18 +361,15 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(true);
             scroll.writeTo(out);
         }
-        out.writeBytesReference(source);
-        out.writeBytesReference(extraSource);
+        if (source == null) {
```

Comment:
can the source ever be null? I think it shouldn't be optional in readFrom and writeTo